### PR TITLE
Add exhaustruct lint

### PR DESCRIPTION
**Pull request by sharnoff**
I know this is generally unpopular in the go world. I think they're probably right in the general case. For this repo, however, given the prevalence of big state objects (and the number of bugs I've run into where I forgot a field), I feel it's better to have this strict exhaustiveness checking here.

The regexes for types to exclude are intended to be added to whenever necessary.

This already caught a couple things that would have been hard to detect otherwise. Looks like most old, high-usage code (i.e. stuff in `pkg/plugin`) was already fine.

Haven't yet run tests to double-check everything's ok.

### CODE REVIEW

File: pkg/plugin/plugin.go [hunk 1]
```diff
@@ -68,7 +68,7 @@ func NewAutoscaleEnforcerPlugin(obj runtime.Object, h framework.Handle) (framewo
 		handle:   h,
 		vmClient: vmClient,
 		// fields are set by p.setConfigAndStartWatcher, p.readClusterState
-		state: pluginState{},
+		state: pluginState{}, //nolint:exhaustruct // see above.
```
Comment (tychoish) [inline pkg/plugin/plugin.go:71]: heh.

**Review by tychoish — COMMENTED** [review]
as previously discussed i'm not sure I buy the enduring utility of this, but this isn't my hill

File: pkg/plugin/state.go [hunk 2]
```diff
@@ -520,6 +526,7 @@ func buildInitialNodeState(node *corev1.Node, conf *config) (*nodeState, error)
 			marginMemory:     marginMemory,
 		},
 		computeUnit: &nodeConf.ComputeUnit,
+		mq:          migrationQueue{},
```
[the remark this answers is not in the document]
Comment (sharnoff) [reply pkg/plugin/state.go:529]: Hah- it is actually exhaustive. `migrationQueue` is defined as `type migrationQueue []*someOtherType` - bit funky, but hey :)

[the remark this answers is not in the document]
Comment (tychoish) [reply pkg/plugin/state.go:529]: ah yeah, I wouldn't do this, for exactly this reason. 

relatedly slices make bad queues.

[the remark this answers is not in the document]
Comment (sharnoff) [reply pkg/plugin/state.go:529]: > relatedly slices make bad queues.

Yep - it's using `container/heap`, which works on top of slices.

**Pull request merged**